src/app/components/grid/client.service.ts

```diff
@@ -75,14 +75,81 @@ type ClearNotificationsType =
 
 interface ClearNotificationsPayload {
 	type: ClearNotificationsType;
-	data: any;
+	data: ClearNotificationsData;
 	clientId: string;
 }
 
+interface ClearNotificationsData {
+	channelId?: number;
+	communityId?: number;
+}
+
 export interface ClearNotificationsEventData extends ClearNotificationsPayload {
 	currentCount: number;
 }
 
+function clearNotifications(type: ClearNotificationsType, data: ClearNotificationsData = {}) {
+	switch (type) {
+		case 'activity':
+			store.commit('setNotificationCount', {
+				type: 'activity',
+				count: 0,
+			});
+			break;
+		case 'notifications':
+			store.commit('setNotificationCount', {
+				type: 'notifications',
+				count: 0,
+			});
+			break;
+		case 'community-channel':
+			{
+				const communityChannelId = data.channelId as number;
+				const communityId = data.communityId as number;
+				const communityState = store.state.communityStates.getCommunityState(communityId);
+				communityState.markChannelRead(communityChannelId);
+			}
+			break;
+		case 'community-featured':
+			{
+				const communityId = data.communityId as number;
+				const communityState = store.state.communityStates.getCommunityState(communityId);
+				communityState.hasUnreadFeaturedPosts = false;
+			}
+			break;
+		case 'community-unread':
+			{
+				const communityId = data.communityId as number;
+				const communityState = store.state.communityStates.getCommunityState(communityId);
+				communityState.hasUnreadPosts = false;
+			}
+			break;
+		case 'friend-requests':
+			// This event gets fired every time the user accepts/rejects a friendship.
+			// The only action we need to take here is to decrease the global friendship number by 1.
+			store.commit('changeFriendRequestCount', -1);
+			break;
+	}
+}
+
+/**
+ * List of resolvers waiting for grid to connect.
+ * These resolvers get resolved in the connect function once Grid connected.
+ */
+let connectionResolvers: (() => void)[] = [];
+/**
+ * Resolves once Grid is fully connected.
+ */
+function tillConnection(client: GridClient) {
```

Comment:
Can't we use just one promise instead of an array? We're resolving all of them with the same exact value essentially in this function and the grid bootstrap resolver

Reply:
I think it's more easily understandable to keep a collection of resolvers and then resolve them all then to respond with one object for everybody to wait on and then resolve it. 
I don't really think it makes a huge difference in terms of usability or performance, so I'd go with this option.